stocks2.lua

```diff
@@ -0,0 +1,544 @@
+    if keys["STRING_A042"] then
+        local new_max_qual = max_qual + 1
+        if new_max_qual > 6 then new_max_qual = 6 end
+        if new_max_qual ~= max_qual then
+            max_qual = new_max_qual
+            refresh_list = true
+        end
+    end
+
+    if keys["CUSTOM_SHIFT_W"] then
+        min_wear = (min_wear + 1) % 4
+        refresh_list = true
+    end
+
+    -- ACTIONS --------------------
+    if keys["CUSTOM_SHIFT_D"] then
+        if apply_to_mode == 1 then --selected
+            _, choice = self.subviews.item_list:getSelected()
+            choice.ref.flags.dump = not choice.ref.flags.dump
+        else
+            for i,v in ipairs(self.subviews.item_list:getChoices()) do
+                v.ref.flags.dump = not v.ref.flags.dump
+            end
+        end
+
+        refresh_list = true
+    end
+
+    if keys["CUSTOM_SHIFT_F"] then
+        if apply_to_mode == 1 then --selected
+            _, choice = self.subviews.item_list:getSelected()
+            choice.ref.flags.forbid = not choice.ref.flags.forbid
+        else
+            for i,v in ipairs(self.subviews.item_list:getChoices()) do
+                v.ref.flags.forbid = not v.ref.flags.forbid
+            end
+        end
+
+        refresh_list = true
+    end
+
+    if keys["CUSTOM_SHIFT_M"] then
+        if apply_to_mode == 1 then --selected
+            _, choice = self.subviews.item_list:getSelected()
+            if is_valid_melt_item(choice.ref) then
+                toggle_melt(choice.ref)
+            end
+        else
+            for i,v in ipairs(self.subviews.item_list:getChoices()) do
+                toggle_melt(v.ref)
+            end
+        end
+
+        refresh_list = true
+    end
+
+
+    if refresh_list then
+        self:filter_list(current_search_query)
+    end
+
+    if keys["CUSTOM_SHIFT_A"] then
+        apply_to_mode = (apply_to_mode + 1) % 2
+        self.subviews.apply_to_label:setText(
+            {{text = "Shift-A", pen = COLOR_LIGHTGREEN}, {text = ": Apply to: ", pen = COLOR_WHITE}, {text = apply_to_labels[apply_to_mode + 1], pen = COLOR_BROWN},
+        })
+    end
+
+    self:show_filters()
+    self:show_qual_and_wear()
+
+    return StocksWindow.super.onInput(self, keys)
+end
+
+function StocksWindow:show_qual_and_wear()
+    local quality_labels = {"Ordinary", "WellCrafted", "FinelyCrafted", "Superior", "Exceptional", "Masterful", "Artifact"}
+    self.subviews.filters_qual_wear:setText(
+        {
+            { text = "-+", pen = COLOR_LIGHTGREEN}, { text = ": Min Qual: ", pen = COLOR_WHITE }, { text = quality_labels[min_qual + 1], pen = COLOR_BROWN }, NEWLINE,
+            { text = "/*", pen = COLOR_LIGHTGREEN}, { text = ": Max Qual: ", pen = COLOR_WHITE }, { text = quality_labels[max_qual + 1], pen = COLOR_BROWN }, NEWLINE,
+            { text = "Shift-W", pen = COLOR_LIGHTGREEN}, { text = ": Min Wear: ", pen = COLOR_WHITE }, { text = tostring(min_wear), pen = COLOR_BROWN }
+        }
+    )
+end
```

Comment:
This should be handed by the RangeSlider

Reply:
You can copy the condition and wear sliders from here: https://github.com/DFHack/scripts/blob/master/internal/caravan/common.lua#L124